internal/prometheus/prometheus.go

```diff
@@ -37,25 +37,14 @@ type Provider struct {
 type loggerFunc func(attrs map[string]interface{}, f string, a ...interface{})
 
 // New returns a new Provider object.
-func New(inner metrics.Metrics, logger loggerFunc) *Provider {
+func New(inner metrics.Metrics, logger loggerFunc, httpRequestBuckets []float64) *Provider {
```

Comment:
Instead of updating the `New` method, we can have a new constructor like `WithHTTPRequestBuckets` for ex.

Reply:
if I were to keep `New` then it would need to call `WithHTTPRequestBuckets` with the default values.

I prefer to keep these default values closed to the config and let the config decide if the default values needs to be used vs the user defined values.

what I did is the rename the `New` with `WithHTTPRequestBuckets` to express the intent.
Let me know please if this works for you.